src/main/java/org/frc5687/infiniterecharge/robot/commands/AutoShoot.java

```diff
@@ -0,0 +1,46 @@
+package org.frc5687.infiniterecharge.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import org.frc5687.infiniterecharge.robot.Constants;
+import org.frc5687.infiniterecharge.robot.OI;
+import org.frc5687.infiniterecharge.robot.subsystems.Indexer;
+import org.frc5687.infiniterecharge.robot.subsystems.Shooter;
+import org.frc5687.infiniterecharge.robot.subsystems.Turret;
+
+public class AutoShoot extends Shoot {
+
+    private long _delayMillis;
+    private long _endMillis = 0;
+
+    public AutoShoot(Shooter shooter, Indexer indexer, Turret turret, OI oi, double speed) {
+        super(shooter, indexer, turret, oi, speed);
+    }
+
+    @Override
+    public void initialize() {
+        _delayMillis = System.currentTimeMillis() + Constants.Auto.AUTO_SHOOT_DELAY;
+    }
+
+    @Override
+    public void execute() {
+
+        if (System.currentTimeMillis() > _delayMillis) {
+            if (_endMillis==0) {
+                super.initialize();
+                _endMillis = System.currentTimeMillis() + Constants.Auto.AUTO_SHOOT_RUNON;
+            }
+            super.execute();
+        }
+
+    }
```

Comment:
Calling `super.initialize()` here from `execute()` feels super weird to me, as does the extra state kept around in `_endMillis`... not gonna "not approve" the PR or anything but really hoping this will get a little simpler when we base this on RPM instead of a fixed time delay.